test/AppNotificationBuilderTests/APITests.cpp

```diff
@@ -382,5 +382,118 @@ namespace Test::AppNotification::Builder
+
+            VERIFY_ARE_EQUAL(builder.BuildNotification().Payload(), expected);
+        }
+
+        TEST_METHOD(AppNotificationBuilderAddTooManyComboBox)
+        {
+            VERIFY_THROWS_HR(AppNotificationBuilder()
+                .AddTextBox(L"input1")
+                .AddTextBox(L"input2")
+                .AddTextBox(L"input3")
+                .AddComboBox(AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox2")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox3")
+                    .AddItem(L"item1", L"item1 text")), E_INVALIDARG);
+        }
+
+        TEST_METHOD(AppNotificationBuilderAddTooManyInputElements)
+        {
+            VERIFY_THROWS_HR(AppNotificationBuilder()
+                .AddComboBox(AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox2")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox3")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox4")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox5")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox6")
+                    .AddItem(L"item1", L"item1 text")), E_INVALIDARG);
+        }
+
+        TEST_METHOD(AppNotificationComboBoxAddTooManySelectionItems)
+        {
+            VERIFY_THROWS_HR(AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text")
+                    .AddItem(L"item2", L"item2 text")
+                    .AddItem(L"item3", L"item3 text")
+                    .AddItem(L"item4", L"item4 text")
+                    .AddItem(L"item5", L"item5 text")
+                    .AddItem(L"item6", L"item6 text"), E_INVALIDARG);
+        }
+
+        TEST_METHOD(AppNotificationComboBoxAddFiveSelectionItems)
+        {
+            auto comboBox{ AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text")
+                    .AddItem(L"item2", L"item2 text")
+                    .AddItem(L"item3", L"item3 text")
+                    .AddItem(L"item4", L"item4 text")
+                    .AddItem(L"item5", L"item5 text") };
+            auto expected{ L"<input id='comboBox1' type='selection'><selection id='item1' content='item1 text'/><selection id='item2' content='item2 text'/><selection id='item3' content='item3 text'/><selection id='item4' content='item4 text'/><selection id='item5' content='item5 text'/></input>" };
+
+            VERIFY_ARE_EQUAL(comboBox.as<winrt::Windows::Foundation::IStringable>().ToString(), expected);
```

Comment:
Recommend using builder.BuildNotification to test developer flow.

Reply:
Some tests do just that but they don;t all need to do that. It just makes the tests more complex and harder to read / maintain.